tree/ntuple/v7/src/RColumnElement.cxx

```diff
@@ -58,6 +58,9 @@ ROOT::Experimental::Detail::RColumnElementBase::Generate<void>(EColumnType type)
    return nullptr;
 }
 
+template std::unique_ptr<ROOT::Experimental::Detail::RColumnElementBase>
+ROOT::Experimental::Detail::RColumnElementBase::Generate<void>(EColumnType type);
+
```

Comment:
@jalopezg-git @jblomer Clang complains about this:
```
/home/jhahnfel/ROOT/src/tree/ntuple/v7/src/RColumnElement.cxx:62:49: warning: explicit instantiation of 'Generate<void>' that occurs after an explicit specialization has no effect [-Winstantiation-after-specialization]
ROOT::Experimental::Detail::RColumnElementBase::Generate<void>(EColumnType type);
                                                ^
/home/jhahnfel/ROOT/src/tree/ntuple/v7/inc/ROOT/RColumnElement.hxx:597:57: note: previous template specialization is here
std::unique_ptr<RColumnElementBase> RColumnElementBase::Generate<void>(EColumnType type);
                                                        ^
1 warning generated.
```

I think this points to a real problem: If a TU only includes `RColumnElement.hxx` and instantiates `Generate<void>`, isn't it an ODR violation to define a different implementation here in `RColumnElement.cxx`?

Reply:
> I think this points to a real problem: If a TU only includes `RColumnElement.hxx` and instantiates `Generate<void>`, isn't it an ODR violation to define a different implementation here in `RColumnElement.cxx`?

Actually not, the template specialization for `CppT = void` is declared in `RColumnElement.hxx`, but only defined in `RColumnElement.cxx`.  Any translation unit including `RColumnElement.hxx` should see the symbol as `extern`.

I created https://github.com/root-project/root/pull/12280; that should fix it.  :-) - my fault, anyways, as I suggested that [here](https://github.com/root-project/root/pull/12133#discussion_r1091805089).

Reply:
Ah right, there's the declaration in the header, makes sense.